docs/README.md

```diff
@@ -648,18 +648,43 @@ capabilities!
 
 Two patterns influence the Booster's event-driven architecture: Command-Query Responsibility Segregation ([CQRS](https://www.martinfowler.com/bliki/CQRS.html)) and [Event Sourcing](https://martinfowler.com/eaaDev/EventSourcing.html). They're complex techniques to implement from scratch with lower-level frameworks, but Booster makes them feel natural and very easy to use.
 
-![architecture][booster-arch-png]
+![architecture](./img/booster-arch.png)
 
-The public interface of a Booster application is just `Commands` and `ReadModels`. Booster proposes an entirely different approach to the Model-View-* and CRUD frameworks. With Booster, the clients submit commands, query the read models, or subscribe to them for receiving real-time updates thanks to the out of the box [GraphQL API][booster-graphql-api].
+The public interface of a Booster application is just `Commands` and `ReadModels`. Booster proposes an entirely different approach to the Model-View-* and CRUD frameworks. With Booster, the clients submit commands, query the read models, or subscribe to them for receiving real-time updates thanks to the out of the box [GraphQL API](#graphql-api)
 
 Booster applications are event-driven and event-sourced so, **the source of truth is the whole history of events**. When a client submits a command, the `CommandHandler` _wakes up_ and executes its logic. Optionally, it can *register* as many `Events` as needed. The framework caches the current state by automatically *reducing* all the registered events into `Entities`. Interested parties can *react* to events via `EventHandlers`, and finally, the *projection* functions transform the entities into `ReadModels`.
 
 In this chapter you'll walk through these concepts and its details.
 
-[booster-graphql-api]:#graphql-api
-[booster-arch-png]:./img/booster-arch.png
+### 1. Command and command handlers
 
-### Event handlers
+#### Commands naming convention
```

Comment:
@javiertoledo commented on my PR that maybe setting naming convention is not necessary

Reply:
I think we should propose one way of doing things and then let people use the framework however they want, or they think it makes sense.

@tainguyenbui @javiertoledo, think about [shuhari](https://en.wikipedia.org/wiki/Shuhari), the Japanese martial art concept which describes the stages of learning to mastery.

- **shu** -> you obey what you have been told because you have no idea of anything
- ha -> you digress the tradition and breaks the rules because you have enough understanding
- ri -> you are already a master, so you don't need rules and tradition because you create them

At this point, we are addressing people in the first stage (everyone, except some of us, is in this stage because Booster is something new for most people). They might need our guidance on how to do it well, and then they can start using the framework breaking the rules.

Maybe we don't need a naming convention for every section, but in commands, it makes sense because the tradition up to this moment is to create five methods in a controller class, and with Booster you have tons of choices.

Reply:
For commands, we could say that we put them in the `commands` folder and use names that are imperative verbs + the object of that verb like `CreateOrder` or `AddItemToCart`. I didn't like the name convention proposed for event handlers because they're already in a folder named `event-handlers` and annotated with the `@EventHandler` decorator, so adding the prefix `Handle` seemed redundant to me. The problem with event handlers is that we initially designed them in Haskell, where you can define many functions with the same name that then you can match by the actual parameter type:

```haskell
handle :: SomeEvent -> [SomeOtherEvent]
handle event = ...
```

and other handler as:
```haskell
handle :: YetAnotherEvent -> [EvenAnotherEvent]
handle event = ...
```

With this design, event handlers don't need a wrapping class or even a name, they're just transformations when an event of kind A appears, I'll emit a list of events of kind B. You don't even need to define which handler you use because the language does the match for you.

With TypeScript, we're forced to wrap handlers in a class to use decorators, and classes require names. I don't know if we can come up with a convention that makes sense for all situations, so I'd let the developer decide. Maybe I'm wrong and it makes a lot of sense to prefix all handlers with `Handle*`, as @tainguyenbui suggested, but I'm not 100% sure.